### PR TITLE
[Debug] Add OutOfMemoryFatalErrorHandler

### PR DESCRIPTION
**Pull request by error56**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | yes     
| Deprecations? | no 
| Tests pass?   | n/a    
| Fixed tickets | #29663  
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+class OutOfMemoryFatalErrorHandler implements FatalErrorHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handleError(array $error, FatalErrorException $exception)
+    {
+        if (false !== strpos($error['message'], 'memory')) {
+            $message = 'I didn\'t get enough memory for execution :\/';
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php]: no need for the $message variable + let's improve the message itself, we never use `I` there :)

File: src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * @author error56 <nalldev2@gmail.com>
+ */
+class OutOfMemoryFatalErrorHandler implements FatalErrorHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handleError(array $error, FatalErrorException $exception)
+    {
+        if (false !== strpos($error['message'], 'memory')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php]: can we be more specific?

Comment (error56) [reply src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php]: Yeah, I will repair it :)

File: src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php (hunk 1 above)
Comment (error56) [reply src/Symfony/Component/Debug/FatalErrorHandler/OutOfMemoryFatalErrorHandler.php]: ok

**Comment by nicolas-grekas** [comment]
I'm not sure I understand when this makes a difference: `OutOfMemoryException` are already handled in a special way in `ErrorHandler` and they *don't* hit fatal error handler.
@OneEyedSpaceFish could you hint about this path also: does it change anything for you? if not, what is your issue about? can you provide a patch?

**Comment by xabbuh** [comment]
@error56 Can you answer @nicolas-grekas' questions? :)

**Comment by error56** [comment]
Yeah, @nicolas-grekas it was a bugfix, IMO this exception would be handled in another way (it's very important exception).
Another questions is to @OneEyedSpaceFish

**Comment by Simperfit** [comment]
@OneEyedSpaceFish could you please check this ?

**Comment by nicolas-grekas** [comment]
I'm sorry I don't understand when this makes a difference. The only case where it could to me is when exception handling is used without passing through `ErrorHandler::handleFatalError()`. I'm not sure this is something that we should support. Please provide a reproducer app if you can. I may miss something and this would help.

**Comment by OneEyedSpaceFish** [comment]
@Simperfit  @nicolas-grekas 

Wow, I'm very late to the party; sorry boys I had missed this completely.
At the top of my head, the issue at the time were that since there were no handlers for this exception, a generic 500 response was returned.

The previous mechanism, has fired an event and a handler was consuming that, making it possible to alter the response body in out of memory exception scenarios.

I'll have a closer look in the next couple of days.

**Pull request closed without merging**